dvc/repo/add.py

```diff
@@ -1,7 +1,10 @@
 import os
+
+from dvc.scm import scm_context
 from dvc.stage import Stage
 
 
+@scm_context
```

Comment:
Shouldn't we use it in `repo.run()` and `repo.imp()` as well?

Reply:
I guess we could, however, corrupting stage as in test for add does not result in dangling edited .gitignore. Do you think we should also have tests for `run` and `imp`?